File: Arrays & Hashing/python/Group Anagrams  .py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        dic={}
        for i in strs:
            liste = [c for c in i]
            liste.sort()
            s=''.join(liste)
            if s in dic:
                dic[s].append(i)
            else:
                dic[s]=[i]
        liste=[]
        for i in dic :
            liste.append(dic[i])
        return liste

File: Arrays & Hashing/c++/Group Anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<string, vector<string>> dic;
        for (int i = 0; i < strs.size(); i++) {
            string sortedStr = strs[i]; 
            sort(sortedStr.begin(), sortedStr.end());
            dic[sortedStr].push_back(strs[i]);
        }
        vector<vector<string>> res;
        for (auto it = dic.begin(); it != dic.end(); ++it) {
            res.push_back(it->second);
        }
        return res;
    }
};